aten/src/ATen/TensorIndexing.h

```diff
@@ -111,15 +111,12 @@ TORCH_API std::ostream& operator<<(std::ostream& stream, const Slice& slice);
 // `torch.tensor([1, 2])`) | `torch::tensor({1, 2})`
 struct TORCH_API TensorIndex final {
   // Case 1: `at::indexing::None`
```

Comment:
Are we sure these are unneeded too? I don't see default initializers for things like boolean_ on all codepaths?

Reply:
The members are
private:
  int64_t integer_ = 0;
  bool boolean_ = false;
  Slice slice_;
  Tensor tensor_;
  TensorIndexType type_;

POD types have in class initialisers and others use default constructor. So all members are initialised .